### ISSUE TITLE
Using keras and tf.keras has different result

### ISSUE THREAD

**Issue by rubiagatra** [tensorflow/tensorflow#15831]
### System information
- **TensorFlow)**:
- **Linux Ubuntu 16.04**:
- **TensorFlow installed from Binary**:
- **TensorFlow version (1.4)**:
- **Python version 3.6.3**: 

### Describe the problem
I am using tf.keras and keras but it has a different result. I am using the code below for sentiment analysis classification using imdb dataset.
### Source code / logs
```
model = Sequential()
model.add(Embedding(n_unique_words, n_dim, input_length=max_reviw_length))
model.add(Flatten())
model.add(Dense(n_dense, activation='relu'))
model.add(Dropout(dropout))
model.add(Dense(1, activation='sigmoid'))
model.summary() 
```

```
_________________________________________________________________
Layer (type)                 Output Shape              Param #   
=================================================================
embedding_1 (Embedding)      (None, 100, 64)           320000    
_________________________________________________________________
flatten_1 (Flatten)          (None, 6400)              0         
_________________________________________________________________
dense_1 (Dense)              (None, 64)                409664    
_________________________________________________________________
dropout_1 (Dropout)          (None, 64)                0         
_________________________________________________________________
dense_2 (Dense)              (None, 1)                 65        
=================================================================
Total params: 729,729
Trainable params: 729,729
Non-trainable params: 0
__________________________

model.compile(loss='binary_crossentropy', optimizer='adam', metrics=['accuracy'])
model.fit(x_train, y_train, batch_size=batch_size, epochs=epochs, verbose=1, validation_data=(x_valid, y_valid), callbacks=[modelcheckpoint])
```
preprocessing all exact the same

tf.keras result

```
Train on 25000 samples, validate on 25000 samples
Epoch 1/4
25000/25000 [==============================] - 9s - loss: 0.7412 - acc: 0.4990 - val_loss: 0.6932 - val_acc: 0.50000.4
Epoch 2/4
25000/25000 [==============================] - 7s - loss: 0.6932 - acc: 0.5005 - val_loss: 0.6931 - val_acc: 0.5000
Epoch 3/4
25000/25000 [==============================] - 7s - loss: 0.6932 - acc: 0.5001 - val_loss: 0.6931 - val_acc: 0.5000
Epoch 4/4
25000/25000 [==============================] - 7s - loss: 0.6932 - acc: 0.4942 - val_loss: 0.6931 - val_acc: 0.5000

<tensorflow.python.keras._impl.keras.callbacks.History at 0x7fad685e7c18>
```

keras result

```
Train on 25000 samples, validate on 25000 samples
Epoch 1/4
25000/25000 [==============================] - 9s 344us/step - loss: 0.5607 - acc: 0.6900 - val_loss: 0.3622 - val_acc: 0.8404
Epoch 2/4
25000/25000 [==============================] - 8s 308us/step - loss: 0.2849 - acc: 0.8849 - val_loss: 0.3486 - val_acc: 0.8446
Epoch 3/4
25000/25000 [==============================] - 8s 305us/step - loss: 0.1179 - acc: 0.9642 - val_loss: 0.4183 - val_acc: 0.8339
Epoch 4/4
25000/25000 [==============================] - 8s 307us/step - loss: 0.0252 - acc: 0.9960 - val_loss: 0.5202 - val_acc: 0.8345

<keras.callbacks.History at 0x7f560b85d940>


```

**Comment by facaiy** [comment]
Similar with #15026, @hhao89 ?

**Comment by hhao89** [comment]
Yes, I think this is the same problem as https://github.com/tensorflow/tensorflow/issues/15026 , and similarly, keras is shown to have a consistently better performance.

**Comment by budach** [comment]
I am running into the same problem described here and in [#15026](https://github.com/tensorflow/tensorflow/issues/15026). The TensorFlow implementation of keras gives very bad results using the Adam optimizer, I observe a more or less constant loss over the epochs while the loss is decreasing as expected using the standalone keras.

**Comment by tensorflowbutler** [comment]
Nagging Assignee: It has been 14 days with no activity and this issue has an assignee. Please update the label and/or status accordingly.

**Comment by tensorflowbutler** [comment]
Nagging Assignee: It has been 14 days with no activity and this issue has an assignee. Please update the label and/or status accordingly.

**Comment by tensorflowbutler** [comment]
Nagging Assignee @fchollet: It has been 14 days with no activity and this issue has an assignee. Please update the label and/or status accordingly.

**Comment by tensorflowbutler** [comment]
Nagging Assignee @fchollet: It has been 14 days with no activity and this issue has an assignee. Please update the label and/or status accordingly.

**Comment by tensorflowbutler** [comment]
Nagging Assignee @fchollet: It has been 14 days with no activity and this issue has an assignee. Please update the label and/or status accordingly.

**Comment by tensorflowbutler** [comment]
Nagging Assignee @fchollet: It has been 15 days with no activity and this issue has an assignee. Please update the label and/or status accordingly.

**Comment by tensorflowbutler** [comment]
Nagging Assignee @fchollet: It has been 15 days with no activity and this issue has an assignee. Please update the label and/or status accordingly.

**Comment by tensorflowbutler** [comment]
Nagging Assignee @fchollet: It has been 137 days with no activity and this issue has an assignee. Please update the label and/or status accordingly.

**Comment by tensorflowbutler** [comment]
Nagging Assignee @fchollet: It has been 152 days with no activity and this issue has an assignee. Please update the label and/or status accordingly.

**Comment by wangyexiang** [comment]
Similar issue. How to solve this issue？@rubiagatra

**Comment by wangyexiang** [comment]
@budach @facaiy @hhao89

**Comment by ErmiaAzarkhalili** [comment]
> Similar issue. How to solve this issue？@rubiagatra

It's there any solution to resolve this issue?

**Comment by wangyexiang** [comment]
see [https://github.com/tensorflow/tensorflow/issues/15026#issuecomment-513819963](url)

**Comment by yairlev** [comment]
Having the same issue with Keras version `2.3.1` versus tf version `2.1.0-dev20191228` and tf.keras version`2.2.4-tf`

**Comment by SachihiroNakagawa** [comment]
@yairlev 
I had a same issue. And I found the reason that BatchNormalization is different in tf.keras and pure keras.
My environment: tensorflow = 1.14.0, tf.keras = 2.2.4-tf, keras = 2.2.4.

The BatchNormalization in tf.keras has the parameters 'trainable' and 'fused'. But in pure keras, there are no 'trainable' and 'fused'.
I had added parameters trainable=False and fuse=False, then the loss in tf.keras went down.
 
The newer verion pure keras already has 'trainable' and 'fused'. So this different is only in old versions.
I hope this helpful.
